contracts/EvmGolf.sol

```diff
@@ -0,0 +1,71 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+contract EvmGolf {
+    mapping(string => address) private _levels; // level name -> level address
+    mapping(address => uint) private _authors; // author -> level address
+    mapping(address => address) private _solutions; // level -> solution
+    mapping(address => address) private _records; // level -> author
+    mapping(address => uint) private _victories; // author -> number of victories
+
+    function registerLevel(address newLevel) external returns (bool) {
+        const levelABI = [ "function name() external returns (string memory)"];
+        const level = new ethers.Contract(newLevel, levelABI, ethers.getDefaultProvider());
+        
+        // Verify that level name is not already registered
+        if (_levels[level.name()]) { return false; }
```

Comment:
Maybe revert instead of returning false?

Reply:
what do you mean by revert? I never save the level in the map to begin with so there isn't anything to revert?

Reply:
A reversion is a transaction that is mined but throws an exception in the evm. We should talk about this.